client/src/common/components/ButtonSwitch/styles.tsx

```diff
@@ -0,0 +1,97 @@
+import styled from 'styled-components';
+
+import theme from '../../styles/theme';
+import { colorChange } from '../../styles/mixins';
+
+import { StylesProps } from '../../../types';
+
+interface ButtonSwitchStylesProps extends StylesProps {
+  width: string;
+  switchType: string;
+  infoType?: string;
+}
+
+export const ButtonSwitch = styled.button<ButtonSwitchStylesProps>`
+  position: relative;
+  width: ${(props: ButtonSwitchStylesProps) => props.width};
+  height: 3rem;
+  padding: 0 2rem;
+  border-radius: ${theme.shape.borderRadius};
+  ${(props: ButtonSwitchStylesProps) =>
+    colorChange(
+      props,
+      'background',
+      theme.palette.componentBackgroundLight,
+      theme.palette.componentBackgroundDark,
+      'background'
+    )}
+  cursor: pointer;
+
+  &::before {
+    content: '';
+    position: absolute;
+    left: ${(props: ButtonSwitchStylesProps) => {
+      if (props.switchType === 'theme') {
+        return props.theme === 'light' ? '0' : '50%';
+      }
+      return props.infoType === 'forecast' ? '0' : '50%';
+    }};
+    top: 0;
+    width: 50%;
+    height: 100%;
+    background: ${theme.palette.primary.light};
+    border-radius: ${theme.shape.borderRadius};
+    box-shadow: ${theme.shadows[0]};
+    transition: 0.8s ${theme.transitions[0]};
+  }
+
+  svg,
+  span {
+    position: absolute;
+    top: 50%;
+    transform: translate(0, -50%);
+  }
+
+  svg {
+    width: 2rem;
+  }
+
+  svg:first-of-type {
+    left: 0.57rem;
+  }
+
+  svg:last-of-type {
+    right: 0.57rem;
+  }
+
+  span {
+    font-weight: 500;
+    transition: color 1.2s;
+  }
+
+  span:first-of-type {
```

Comment:
why not create unified selector for color like this?
`span:first-of-type, span:last-of-type {  color: }`
`span:first-of-type { left: }`
`span:last-of-type { right: }`

Reply:
Ok, will do.